include/deal.II/base/time_stepping.h

```diff
@@ -833,7 +830,7 @@ namespace TimeStepping
      * If the flag is true, the last stage is the same as the first stage and
      * one evaluation of f can be saved.
      */
-    bool last_same_as_first;
+    bool last_same_as_first = false;
```

Comment:
I could be wrong, but I don't think that we normally do in-place initialisation for variables that aren't static or constexpr. Any chance that this and `last_stage` could just be initialised in the constructor?

Reply:
I could but then I can't use `default` for the constructor since that's how we got the bug. The CPP Core Guidelines recommend in-class initialization to avoid the current bug (see [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c48-prefer-in-class-initializers-to-member-initializers-in-constructors-for-constant-initializers)). 

I can initialize the variable in the constructor if you want me too but I rather not.

Reply:
Ah, I didn't notice that the constructor was marked `default`. Ok, then I'm fine with it how it is. Thanks!

Reply:
@jppelteret is right that we don't usually do this, by the way. This may be a personal taste, but I don't like this syntax because it suggests that this is the value the variable will carry, when in reality it's just the initializer it has at the beginning, which is confusing when you read through the class declaration in my opinion.

It's different if this were a `const` variable.

Reply:
I knew you didn't like it when I wrote this but I do :stuck_out_tongue: I find that better than having to repeat the value for each constructor because sometimes I forget one or I use `default` like here and then we get a bug.

Reply:
Ha :-) We don't always have to agree, I guess, and we're a collective, so my personal style preferences need not win.

I think that the general strategy for these kinds of things are "inheriting constructors" where each constructor would first forward to the default constructor in which you set the default values, and then do whatever else you need to do. This way the initializer for each variable has to be specified only in the default constructor, and not repeated in the other constructors. But I don't think it particularly matters.